index.js

```diff
@@ -325,24 +326,63 @@ if(!helpers.isTestEnvironment()){
 }
 
 // Handler for income SmartThings POST requests
-app.post('/api/st', function(req, res, next) {    // eslint-disable-line no-unused-vars -- next might be used in the future
-    smartapp.handleHttpCallback(req, res);
-});
+app.post('/api/st', Validator.body(['lifecycle']).exists(), function(req, res, next) {    // eslint-disable-line no-unused-vars -- next might be used in the future
+    try {
+        const validationErrors = Validator.validationResult(req)
+  
+        if(validationErrors.isEmpty()){
+            smartapp.handleHttpCallback(req, res)
+        } else {
+            helpers.log(`Bad request, parameters missing ${JSON.stringify(validationErrors)}`)
```

Comment:
Does the validationErrors object already have a nice string to display? I wonder if we even need to put the `Bad request, parameters missing` part around it.

Reply:
I'm not sure, I will set it off in postman and see 😅 for this one the error that comes back is re smartapp looking for an authorisation header that I don't have so I didn't notice it as it doesn't make it to the `else`, but this came from where `Validator` is used in Buttons so I think it might be needed 🤔 I'll check though

Reply:
```2021-01-15T00:14:07.178Z - Bad request, parameters missing {"errors":[{"msg":"Invalid value","param":"lifecycle","location":"body"}]}```

Reply:
I think missing is more descriptive than invalid value personally, especially because lifecycle wants to be `===` to something and it being wrong is different than not existing at all, but am happy to do whatever!

Reply:
(that's what `.exists()` comes back with)

Reply:
Yup, looks like they purposely are not super explicit in their messages:
https://express-validator.github.io/docs/custom-error-messages.html